e2e-tests/cypress/tests/integration/channels/channel/channel_settings_modal_spec.ts

```diff
@@ -0,0 +1,390 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channels @channel @channel_settings
+
+import {Team} from '@mattermost/types/teams';
+import {Channel} from '@mattermost/types/channels';
+import {UserProfile} from '@mattermost/types/users';
+
+describe('Channel Settings Modal', () => {
+    let testTeam: Team;
+    let testUser: UserProfile;
+    let testChannel: Channel;
+    let originalTestChannel: Channel;
+
+    before(() => {
+        // Setup test data
+        cy.apiInitSetup().then(({team, user}) => {
+            testTeam = team;
+            testUser = user;
+
+            // Create a test channel
+            cy.apiCreateChannel(testTeam.id, 'test-channel', 'Test Channel').then(({channel}) => {
+                testChannel = channel;
+                originalTestChannel = {...channel};
+                cy.apiAddUserToChannel(channel.id, user.id);
+            });
+
+            cy.apiLogin(testUser);
+        });
+    });
+
+    beforeEach(() => {
+        // Visit the channel before each test
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+    });
+
+    // 2DO: update tests ids once QA has defined the test plan and zephyr tests
```

Comment:
@lindalumitchell please, review this test suite and let me know if you consider enough this test suite and/or if there should be other scenarios/flows the test should cover. Also, the ids should change once zephyr tests cases are defined, right?